### ISSUE TITLE
too-few-public-methods emitted for dataclasses

### ISSUE THREAD

**Issue by egpbos** [PyCQA/pylint#3025]
### Steps to reproduce
1. Create a file pylint_dataclasses.py containing:

```py
"""Demonstrate pylint false positive R0903"""
import dataclasses
import numpy as np


@dataclasses.dataclass(frozen=True)
class Point:
    """A three dimensional point with x, y and z components."""

    x: float
    y: float
    z: float

    def to_array(self):
        """Convert to a NumPy array `np.array((x, y, z))`."""
        return np.array((self.x, self.y, self.z))
```

2. Run pylint on it:

```sh
pylint pylint_dataclasses.py
```

### Current behavior
```sh
************* Module pylint_dataclasses
pylint_dataclasses.py:6:0: R0903: Too few public methods (1/2) (too-few-public-methods)
```

### Expected behavior
No R0903 emitted, because dataclasses should not be expected to have many methods.

### pylint --version output
```sh
pylint 2.3.1
astroid 2.2.5
Python 3.7.1 (default, Dec 14 2018, 13:28:58)
[Clang 4.0.1 (tags/RELEASE_401/final)]
```

**Comment by onlinejudge95** [comment]
I also faced the same issue, when storing my config for a `flask` app in classes. Pylint reports `R0903: Too few public methods (0/2) (too-few-public-methods)`, even though it makes perfect sense for classes annotated by `dataclasses.dataclass` to have few methods. 

Any updates.

**Comment by PCManticore** [comment]
No updates just yet, feel free to submit a patch.
